dckks/relinkey_gen.go

```diff
@@ -65,6 +71,11 @@ func (ekg *RKGProtocol) NewEphemeralKey(p float64) (ephemeralKey *ring.Poly) {
 func (ekg *RKGProtocol) GenShareRoundOne(u, sk *ring.Poly, crp []*ring.Poly, shareOut RKGShareRoundOne) {
 
 	contextQP := ekg.dckksContext.contextQP
+	prng, err := utils.NewPRNG()
+	if err != nil {
+		panic(err)
+	}
+	gaussianSampler := ring.NewGaussianSampler(prng, contextQP)
```

Comment:
The gaussian sampler could be pre-allocated as a field in the `RKGProtocol` type